### ISSUE TITLE
tapioca's own rbi yields strange type check errors in 0.4.25

### ISSUE THREAD

**Issue by fdr** [Shopify/tapioca#483]
For some reason, as of 0.4.25, I can produce these very strange typechecking problems with tapioca, when it generates its own rbi:

```
sorbet/rbi/gems/tapioca@0.4.25.rbi:496: Unable to resolve constant Specification https://srb.help/5002
     496 |  sig { returns([T::Array[Gem::Specification], T::Array[String]]) }
                                    ^^^^^^^^^^^^^^^^^^
  Autocorrect: Use `-a` to autocorrect
    sorbet/rbi/gems/tapioca@0.4.25.rbi:496: Replace with Gem::Specification
     496 |  sig { returns([T::Array[Gem::Specification], T::Array[String]]) }
                                    ^^^^^^^^^^^^^^^^^^
  Autocorrect: Use `-a` to autocorrect
    sorbet/rbi/gems/tapioca@0.4.25.rbi:496: Replace with Gem::Resolver::Specification
     496 |  sig { returns([T::Array[Gem::Specification], T::Array[String]]) }
                                    ^^^^^^^^^^^^^^^^^^
    https://github.com/sorbet/sorbet/tree/a31ad73a52c5be7d6f5363bf5e88224c5e46e11f/rbi/stdlib/rubygems.rbi#L1315: Did you mean: Gem::Specification?
    1315 |class Gem::Specification < Gem::BasicSpecification
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    https://github.com/sorbet/sorbet/tree/a31ad73a52c5be7d6f5363bf5e88224c5e46e11f/rbi/stdlib/rubygems.rbi#L6456: Did you mean: Gem::Resolver::Specification?
    6456 |class Gem::Resolver::Specification
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

It's strange because the suggestion appears identical to what the source code says. And I cannot produce it in 0.4.24.

I've produced this on Ruby 3 and Ruby 2.7.

**Comment by fdr** [comment]
I chose to use this `config.yml`:

```
exclude:
  - tapioca
```

But why not release a new tapioca? Seems serious.

**Comment by paracycle** [comment]
@fdr you are right, we should make a new stable release. When fixing this problem on the main branch, I'd forgotten that it needed a backport. I can make the backport tomorrow or more likely on Tuesday and make a 0.4.26 release.

**Comment by paracycle** [comment]
Just released [v0.4.26](https://github.com/Shopify/tapioca/releases/tag/v0.4.26) to fix this problem and another one related to some edge-case load ordering issue.

Please let us know if you are still seeing issues. Thanks again for reporting the problem.

**Comment by fdr** [comment]
Thanks, it works beautifully
